python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets.py

```diff
@@ -208,3 +209,96 @@ def test_coerced_asset_keys():
     @asset(ins={"input1": AssetIn(asset_key=["Asset", "1"])})
     def asset1(input1):
         assert input1
+
+
+def test_asset_with_io_manager_def():
+    io_manager_inst = InMemoryIOManager()
```

Comment:
I believe this works, but I think this test would be more thorough if the IOManager instance was some custom IOManager that did something obviously different from the default behavior. Something like
```
class CustomIOManager(IOManager):
    def load_input(self, context):
        return "abcdefg"
```